element-hq/element-x-android
# Hide media preprocessing

## Content


Pre-process the media to upload during the preview so that it can be sent instantly.

2 scenarii:
- the user presses the send buttton, but the pre-processing is not finish yet. In this case the progress dialog is still displayed
- the user presses the sent button and the pre-processing is done, in this case no dialog is displayed.

## Motivation and context


Closes #3938

## Screenshots / GIFs



https://github.com/user-attachments/assets/f62d8dc1-cc9f-479e-ba1d-2806b496bd32

## Tests



- Send a video or an image and wait a bit before pressing the send button
- Press the send button
- Observe that no dialog is displayed and the local echo with the media is displayed in the timeline.

Second test
- Send a video and press the send button just after
- Observe that a dialog is displayed
- When the screen is closed the dialog is first closed (this is the second commit)

## Tested devices

- OS version(s):

## Checklist
